cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/BatchSQLEngineAdapter.java

```diff
@@ -813,6 +813,26 @@ protected static void countStageMetrics(StageMetrics stageMetrics,
     });
   }
 
+  /**
+   * Executes a task based o provided {@link Supplier}. This returns a {@link CompletableFuture} with the result of
+   * the execution of the supplied task.
+   *
+   * @param supplier task to execute.
+   * @param <T> Output type for this job
+   * @return {@link CompletableFuture} with the output for the execution of this task.
+   */
+  public <T> CompletableFuture<T> submitTask(Supplier<T> supplier) {
```

Comment:
Is not it exactly same as `CompletableFuture.supplyAsync(supplier, executorService)`?

Reply:
It is! I was not aware of this function (using the executor as param).

I'll refactor this.